ui/components/search/search.js

```diff
@@ -18,21 +27,41 @@ export class Search extends React.Component {
 
   actionPath() {
     const path = this.props.router.pathname;
-    return (path.includes('policies') || path.includes('requirements')) ? path : '/requirements/';
+    return path.includes('policies') || path.includes('requirements') ? path : '/requirements/';
+  }
+
+  handleChange(e) {
+    const { value } = e.target;
+    this.setState({ term: value });
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const { router } = this.props;
+    const { term } = this.state;
+    const pathname = this.actionPath();
+    const query = { [this.inputName()]: term };
+    router.push({ pathname, query });
   }
 
   render() {
     return (
       <div className="search-form pr2 no-print">
-        <form method="GET" action={this.actionPath()} className="mb0 flex items-center">
+        <form
+          method="GET"
+          action={this.actionPath()}
+          className="mb0 flex items-center"
+          onSubmit={this.handleSubmit}
+        >
           <input
             aria-label="Search term"
             name={this.inputName()}
             type="text"
             placeholder="Search"
             className="search-input p1 gray-border"
+            onChange={this.handleChange}
```

Comment:
In an earlier iteration, we had set the value to `this.state.term`, but I'm not sure if there was a benefit. Is there any risk of "redrawing" or similar that'd lose the user's value?

Reply:
Yes, if we set the value as an attribute from `this.state` (which is what a lot of React docs, community and official, recommend we do) then everytime the event fires, render will be called and the result will be different, resulting in ReactDOM updating the value of the field. There isn't too much of a risk where we'll lose the user's value as the above render sequence is triggered when a user changes the value of the field. However, I have noticed that it can add a small lag to the field interaction which makes the website feel a bit less professional.